### ISSUE TITLE
`StringIO.new` does not accept an `encoding:` keyword argument

### ISSUE THREAD

**Issue by postmodern** [oracle/truffleruby#2793]
`StringIO.new` does not appear to accept keyword arguments.

### TruffleRuby 22.3.0

```ruby
require 'stringio'
StringIO.new(encoding: Encoding::ASCII_8BIT)
```
```
<internal:core> core/type.rb:104:in `coerce_to_failed': wrong argument type Hash (expected String) (TypeError)
	from <internal:core> core/type.rb:93:in `execute_coerce_to'
	from <internal:core> core/type.rb:85:in `coerce_to'
	from /opt/rubies/truffleruby-22.3.0/lib/truffle/stringio.rb:146:in `initialize'
	from (irb):1:in `<top (required)>'
	from <internal:core> core/kernel.rb:407:in `loop'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from /opt/rubies/truffleruby-22.3.0/lib/gems/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from <internal:core> core/kernel.rb:376:in `load'
	from <internal:core> core/kernel.rb:376:in `load'
	from /opt/rubies/truffleruby-22.3.0/bin/irb:42:in `<main>'
<internal:core> core/type.rb:91:in `execute_coerce_to': undefined method `to_str' for {:encoding=>#<Encoding:ASCII-8BIT>}:Hash (NoMethodError)
Did you mean?  to_s
               to_set
	from <internal:core> core/type.rb:85:in `coerce_to'
	from /opt/rubies/truffleruby-22.3.0/lib/truffle/stringio.rb:146:in `initialize'
	from (irb):1:in `<top (required)>'
	from <internal:core> core/kernel.rb:407:in `loop'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from /opt/rubies/truffleruby-22.3.0/lib/gems/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from <internal:core> core/kernel.rb:376:in `load'
	from <internal:core> core/kernel.rb:376:in `load'
	from /opt/rubies/truffleruby-22.3.0/bin/irb:42:in `<main>'
```

```ruby
StringIO.new("foo", encoding: Encoding::ASCII_8BIT)
```
```
<internal:core> core/type.rb:276:in `convert_type': no implicit conversion of Hash into String (TypeError)
	from <internal:core> core/type.rb:258:in `rb_convert_type'
	from <internal:core> core/kernel.rb:171:in `StringValue'
	from /opt/rubies/truffleruby-22.3.0/lib/truffle/stringio.rb:154:in `initialize'
	from (irb):2:in `<top (required)>'
	from <internal:core> core/kernel.rb:407:in `loop'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from /opt/rubies/truffleruby-22.3.0/lib/gems/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from <internal:core> core/kernel.rb:376:in `load'
	from <internal:core> core/kernel.rb:376:in `load'
	from /opt/rubies/truffleruby-22.3.0/bin/irb:42:in `<main>'
```

### CRuby 3.0.4

```ruby
require 'stringio'
StringIO.new(encoding: Encoding::ASCII_8BIT)
# => #<StringIO:0x00005562ac9f45e8>
StringIO.new("foo", encoding: Encoding::ASCII_8BIT)
# => #<StringIO:0x00005562ac82c6e8>
```

**Comment by flavorjones** [comment]
It doesn't seem like this option does anything in CRuby. Using 3.1.2:

```ruby
#! /usr/bin/env ruby

require "stringio"

StringIO.new().string.encoding
# => #<Encoding:UTF-8>

StringIO.new(encoding: Encoding::KOI8_R).string.encoding
# => #<Encoding:UTF-8>

StringIO.new(String.new(encoding: Encoding::KOI8_R)).string.encoding
# => #<Encoding:KOI8-R>
```

**Comment by andrykonchin** [comment]
Looking at the CRuby implementation of StringIO I assume that `StringIO.new` excepts the keyword arguments only for signature compatibility with `IO.new` and `File.new`.  It checks some error cases and just ignores values.

Examples of errors:

```ruby
StringIO.new("", "w", mode: "r")
# (irb):15:in `initialize': mode specified twice (ArgumentError)

StringIO.new("", "w:utf-8", encoding: "ascii-8bit")
# (irb):16:in `initialize': encoding specified twice (ArgumentError)
```

**Comment by andrykonchin** [comment]
Fixed in d4c94d8f97d08f3cd4a069833e8056725eab9878. Thank you for reporting!
